crates/pagecache/src/iobuf.rs

```diff
@@ -128,15 +128,19 @@ impl IoBufs {
             next_lid
         );
 
+        // we want stable to begin at -1 if the 0th byte
+        // of our file has not yet been written.
+        let stable = next_lsn - 1;
```

Comment:
You should use the [`wrapping_sub`](https://doc.rust-lang.org/std/primitive.isize.html#method.wrapping_sub) function to avoid Rust to panic on overflowing arithmetic operations.

Reply:
If you always want to use the `-1` value as a marker, obviously.

Reply:
it's an isize and we want next_lsn of 0 to be -1

Reply:
Ahha! Wooops...